presto-orc/src/main/java/com/facebook/presto/orc/metadata/Footer.java

```diff
@@ -71,6 +87,15 @@ public String toString()
                 .add("stripes", stripes)
                 .add("types", types)
                 .add("columnStatistics", fileStats)
+                .add("userMetadata", getUserMetadataAsString())
                 .toString();
     }
+
+    private String getUserMetadataAsString()
+    {
+        return userMetadata.entrySet().stream().collect(toMap(
+                Map.Entry::getKey,
+                entry -> entry.getValue().toStringUtf8()
```

Comment:
This will throw if the data is not valid UTF-8. It could use `fixInvalidUtf8()` first, but I'm not sure UTF-8 is the correct thing to do here given that metadata can be arbitrary binary data. I'm thinking we could either

1. Use `length=XXX` for the value, similar to `Slice`
2. Omit it from `toString`